### ISSUE TITLE
LTI 1.3: Institutional Admin roles should have the same permissions as teachers

### ISSUE THREAD

**Issue by chrispittman** [longhornopen/qualtrics-lti#14]
See details at https://github.com/longhornopen/qualtrics-lti/issues/13#issuecomment-1502161921

**Comment by jonespm** [comment]
I'm _guessing_ it's something in the changes in the PHP library, possibly with this larger recent change. https://github.com/celtic-project/LTI-PHP/commit/3f62ab78850438fda2de5139e6cf97c3ab21e7f3

With LTI 1.3 this was working for admins in release 20230215 but broken in 20230404.

The roles that are being sent for an admin are

```
  "https://purl.imsglobal.org/spec/lti/claim/roles": [
    "http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator",
    "http://purl.imsglobal.org/vocab/lis/v2/institution/person#Instructor",
    "http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student",
    "http://purl.imsglobal.org/vocab/lis/v2/system/person#User"
  ],
```

I'm not sure how this is parsed by the library. It looks like it should be using these for LTI 1.3, and not some Canvas roles substitution. 

If a user has the institution role `http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator` then they are admin. 

Perhaps it has something to do with this third parameter here of `$addPrincipalRole` which defaults to false? I'm not seeing code that's flipping that. 

If this was made configurable (to either allow or not allow admins the same role as instructor) it seems like that would also work.

**Comment by chrispittman** [comment]
Think I've identified the core problem here in an upstream library, and have opened an issue against the library to see if we can get a fix from upstream.

https://github.com/celtic-project/LTI-PHP/issues/53

**Comment by chrispittman** [comment]
The celtic/lti library hasn't released a version with this fix yet, but I've gone ahead and pull their fix into our upstream fork manually.  This should be fixed as of Docker image 20230421, or commit 5847dda.
